esm/shared/attributes.js

```diff
@@ -89,6 +89,23 @@ export const stringAttribute = {
   }
 };
 
+export const urlAttribute = {
+  get(element, name) {
+    let attr = element.getAttribute(name);
+    if (attr) {
+      try {
+        return new URL(attr, element.baseURI).href;
```

Comment:
in which case this could fail, if I might ask? 🤔

Reply:
I guess my wonder is why this wouldn't be enough:

```js
export const urlAttribute = {
  get(element, name) {
    const attr = element.getAttribute(name);
    return attr ? new URL(attr, element.baseURI).href : '';
  },
  set(element, name, value) {
    element.setAttribute(name, value);
  }
};
```

but for unknown crawling maybe your version is safer ... although is the result expected at that point?